rpc/errors.go

```diff
@@ -18,6 +18,11 @@ package rpc
 
 import "fmt"
 
+var (
+	// ErrConnectionClosed is thrown when the connection is closed
+	ErrConnectionClosed = new(connectionClosedError)
```

Comment:
Is an entire new type really needed? Can't we just do `errors.New` here? If no, I'd rather make the type a tad more generic because currently the type itself is constant too (i.e. code and error message not configurable).